charts/immich/values.yaml

```diff
@@ -1,268 +1,133 @@
----
+# These entries are shared between all the Immich components
+
+env:
+  REDIS_HOSTNAME: '{{ printf "%s-redis-master" .Release.Name }}'
+  DB_HOSTNAME: "{{ .Release.Name }}-postgresql"
+  DB_USERNAME: "{{ .Values.postgresql.global.postgresql.auth.username }}"
+  DB_DATABASE_NAME: "{{ .Values.postgresql.global.postgresql.auth.database }}"
+  # -- You should provide your own secret outside of this helm-chart and use `postgresql.global.postgresql.auth.existingSecret` to provide credentials to the postgresql instance
+  DB_PASSWORD: "{{ .Values.postgresql.global.postgresql.auth.password }}"
+  TYPESENSE_ENABLED: "{{ .Values.typesense.enabled }}"
+  TYPESENSE_API_KEY: "{{ .Values.typesense.env.TYPESENSE_API_KEY }}"
+  TYPESENSE_HOST: '{{ printf "%s-typesense" .Release.Name }}'
+  IMMICH_WEB_URL: '{{ printf "http://%s-web:3000" .Release.Name }}'
+  IMMICH_SERVER_URL: '{{ printf "http://%s-server:3001" .Release.Name }}'
+  IMMICH_MACHINE_LEARNING_URL: '{{ printf "http://%s-machine-learning:3001" .Release.Name }}'
+
+image:
+  tag: v1.52.1
+
+immich:
+  persistence:
+    # Main data store for all photos shared between different components.
+    library:
+      # Automatically creating the library volume is not supported by this chart
+      # You have to specify an existing PVC to use
+      existingClaim:
+
+# Dependencies
+
 postgresql:
   enabled: false
   global:
     postgresql:
       auth:
         username: immich
         database: immich
+        password: immich
 
 redis:
   enabled: false
   architecture: standalone
   auth:
     enabled: false
 
-common_env:
-  REDIS_HOSTNAME: '{{ printf "%s-redis-master" (include "common.names.fullname" .) }}'
-  DB_HOSTNAME: '{{ include "common.names.name" . }}-postgresql'
-  DB_USERNAME: "{{ .Values.postgresql.global.postgresql.auth.username }}"
-  DB_DATABASE_NAME: "{{ .Values.postgresql.global.postgresql.auth.database }}"
-  JWT_SECRET: "randomstringthatissolongandpowerfulthatnoonecanguess"
-  # ENABLE_MAPBOX is either true of false -> if true, you have to provide MAPBOX_KEY
-  ENABLE_MAPBOX: false
-  MAPBOX_KEY: ""
-  # -- You should provide your own secret outside of this helm-chart and use `postgresql.global.postgresql.auth.existingSecret` to provide credentials to the postgresql instance
-  DB_PASSWORD: "thisshouldnotbeyourpassword"
+typesense:
+  enabled: false
+  env:
+    TYPESENSE_DATA_DIR: /tsdata
+    TYPESENSE_API_KEY: typesense
+  persistence:
+    tsdata:
```

Comment:
Shall we change this to `enabled: true` and by default `type: emptyDir` with proper `mountPath` option?

Reply:
The postgres data volume is generated by postgres sts. Maybe make typesense to be a sts is a good idea.

Reply:
Using an sts also interacts with the DNS hostname and such, right? And I don't really see the difference between defining a PVC vs letting it be generated by the sts. I don't fully know the intricacies, but I don't think we need an sts here.

The mountPath for volumes generated by the common-library is derived from the name of the volume, eg:
```yaml
persistence:
  tsdata: # <- mountPath: tsdata
    etc: ...
```

Reply:
> Using an sts also interacts with the DNS hostname and such, right? 

Yes, In terms of dns resolution, it is the same as that of Deployment.
I thought the `tsdata` volume should be keep even after `helm uninstall`. But as you said, it can be regenerated from posrgres, so there is not necessary to retain it.

> The mountPath for volumes generated by the common-library is derived from the name of the volume

Should we give this an explanation?

Reply:
I'll add a comment referencing the common-library's values.yaml as the place to go for more in-depth information about how things work.